### PR TITLE
Middleware cancellation token propagation

### PR DESCRIPTION
**Pull request by gkinsman**

### CODE REVIEW

File: src/JustSaying/JustSayingBus.cs [hunk 1]
```diff
@@ -165,6 +165,8 @@ private Task RunImpl(CancellationToken stoppingToken)
 
             SubscriptionGroups = subscriptionGroupFactory.Create(_subscriptionGroupSettings);
 
+            _log.LogDebug("Interrogation response: {@Response}", SubscriptionGroups.Interrogate());
```
Comment (maisiesadler) [inline src/JustSaying/JustSayingBus.cs:168]: Check if debug log is enabled before calling SubscriptionGroups.Interrogate()?

File: src/JustSaying/JustSayingFluently.cs [hunk 2]
```diff
@@ -251,6 +251,7 @@ public IFluentSubscription IntoQueue(string queueName)
 
             _subscriptionConfig.TopicName = GetOrUseTopicNamingConvention<T>(_subscriptionConfig.TopicName);
             _subscriptionConfig.QueueName = GetOrUseQueueNamingConvention<T>(_subscriptionConfig.QueueName);
+            _subscriptionConfig.SubscriptionGroup ??= _subscriptionConfig.QueueName;
```
Comment (maisiesadler) [inline src/JustSaying/JustSayingFluently.cs:254]: This seems odd 🤔

Comment (gkinsman) [reply src/JustSaying/JustSayingFluently.cs:254]: So this is setting the default - if no subscription group is set by the fluent API, then we set it to the queue name. Is it the syntax you find weird or that we're doing this here?

Comment (maisiesadler) [reply src/JustSaying/JustSayingFluently.cs:254]: I found it weird that `SubscriptionGroup` was being set to `QueueName`, felt like the types should mismatch here. Is SubscriptionGroup just the name of the group?

Comment (gkinsman) [reply src/JustSaying/JustSayingFluently.cs:254]: yeah it is - perhaps it should be called `SubscriptionGroupName` inline with the other variables. It's the name attached to the read config in the fluent API that we use later to link the queue to the subscription group.

Comment (maisiesadler) [reply src/JustSaying/JustSayingFluently.cs:254]: i think SubscriptionGroupName makes more sense 👍

**Pull request merged**